yasp.c

```diff
@@ -45,16 +46,13 @@ int16_t process_yasp(uint8_t *buffer, uint16_t length)
         return RET_CMD_NOSYNCH;
     }
     cmd_len = (((uint16_t) buffer[LENGTH_POS]) << 8) + ((uint16_t)buffer[LENGTH_POS+1]);
-    if (cmd_len > (length - START_CHECKSUM)) // Synch not included in command length
+    if (cmd_len > (length - START_CRC32)) // Synch not included in command length
     {
         return RET_CMD_INCOMPLETE;
     }
-    for (i = START_CHECKSUM; i < cmd_len + START_CHECKSUM; i++)
-    {
-        checksum += buffer[i];
-    }
-    //printf("Actual: %02x, Checksum: %02x\n", buffer[cmd_len + START_CHECKSUM], checksum);
-    if (checksum != buffer[cmd_len + START_CHECKSUM])
+    crc_32 = (crc_32, &buffer[START_CRC32], cmd_len);
```

Comment:
What is going on here?  Shouldn't this be crc32(crc_32, &buffer[START_CRC32], cmd_len)?